core/src/test/java/org/apache/accumulo/core/fate/TestStore.java

```diff
@@ -32,25 +32,54 @@
 import java.util.function.Consumer;
 import java.util.stream.Stream;
 
+import org.apache.accumulo.core.util.Pair;
+
+import com.google.common.hash.HashCode;
+import com.google.common.hash.Hashing;
+
 /**
  * Transient in memory store for transactions.
  */
 public class TestStore implements FateStore<String> {
 
   private long nextId = 1;
-  private Map<FateId,TStatus> statuses = new HashMap<>();
-  private Set<FateId> reserved = new HashSet<>();
-
+  private final Map<FateId,Pair<TStatus,Optional<FateKey>>> statuses = new HashMap<>();
+  private final Map<FateId,Map<Fate.TxInfo,Serializable>> txInfos = new HashMap<>();
+  private final Set<FateId> reserved = new HashSet<>();
   private static final FateInstanceType fateInstanceType = FateInstanceType.USER;
-  private Map<FateId,Map<Fate.TxInfo,Serializable>> txInfos = new HashMap<>();
 
   @Override
   public FateId create() {
     FateId fateId = FateId.from(fateInstanceType, nextId++);
-    statuses.put(fateId, TStatus.NEW);
+    statuses.put(fateId, new Pair<>(TStatus.NEW, Optional.empty()));
     return fateId;
   }
 
+  @Override
+  public Optional<FateTxStore<String>> createAndReserve(FateKey key) {
+    HashCode hashCode = Hashing.murmur3_128().hashBytes(key.getSerialized());
```

Comment:
This TestStore is only used by unit test. If the existing unit test does not call thsi method then could throw an UnsupportedOpException.